okhttp/src/test/java/okhttp3/internal/http2/HttpOverHttp2Test.java

```diff
@@ -104,11 +110,13 @@
     return asList(Protocol.H2_PRIOR_KNOWLEDGE, Protocol.HTTP_2);
   }
 
-  @Rule public final PlatformRule platform = new PlatformRule();
+  private JdkMatchRule jdkMatchRule = new JdkMatchRule();
+  private PlatformRule platform = new PlatformRule();
+  @Rule public final TestRule chain =
```

Comment:
Don’t they chain on their own?

Reply:
Predictably and stable, yes!? But just not intuitively e.g. field ordering.  I suspect by design if each Rule only affects the test execution you are grand.  But here the Rules can fail e.g. timeout, and we want those failures to be seen or not seen by other rules.